experimental/uefi/app/src/main.rs

```diff
@@ -26,15 +26,12 @@
 
 #[macro_use]
 extern crate log;
+extern crate alloc;
 
-use uefi::{
-    prelude::*,
-    proto::console::serial::Serial,
-    table::{
-        boot::{OpenProtocolAttributes, OpenProtocolParams},
-        runtime::ResetType,
-    },
-};
+use ciborium::{de, ser};
```

Comment:
While this works for now, I don't think CBOR is a suitable framing mechansim for the long term. If we start creating more complex structs to represents frames with headers this will start adding a lot of overhead. AFAICT structs are represented as maps, with each field name as a string key. This will waste bandwith and be slower to serialise and deserialise. (see https://docs.rs/ciborium/latest/ciborium/#packed-encoding)

I think we should implemment the framing manually, as I believe this communication channel is likely to be a performance bottleneck.

Perhaps create and issue and a TODO somewhere to track this.

Reply:
+1 I think at this level we should implement a simple manual serialization, e.g. something like a standard length delimited framing.

Reply:
I'll leave it as-is for now but get back to it later. On the loader side we can use https://docs.rs/tokio-util/latest/tokio_util/codec/length_delimited/index.html; we'd likely need to implement something similar on the UEFI app side.